### PR TITLE
feat(DDM): add ddm schema resource

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add ddm schema resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add ddm schema resource
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=
'./huaweicloud/services/acceptance/ddm/' TESTARGS='-run TestAccDdmSchema_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ddm/ -v -run TestAccDdmSchema_basic -timeout 360m -parallel 4
=== RUN   TestAccDdmSchema_basic
=== PAUSE TestAccDdmSchema_basic
=== CONT  TestAccDdmSchema_basic
--- PASS: TestAccDdmSchema_basic (701.72s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ddm       701.767s
```

### CODE REVIEW

File: huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+				ValidateFunc: validation.StringInSlice([]string{
+					"cluster", "single",
+				}, false),
+			},
+			"shard_number": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ForceNew:     true,
+				Description:  `Specifies the number of shards in the same working mode.`,
+				ValidateFunc: validation.IntBetween(1, 32),
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go]: the valid range is not correct

File: huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+				Description: `Specifies the username for logging in to the associated RDS instance.`,
+			},
+			"admin_password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the password for logging in to the associated RDS instance.`,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go]:
```suggestion
```

File: huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go [hunk 3]
```diff
@@ -0,0 +1,489 @@
...
+				Computed:    true,
+				Description: `Indicates the name of the associated RDS instance.`,
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Optional:    true,
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go]:
```suggestion
```

File: huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go [hunk 4]
```diff
@@ -0,0 +1,489 @@
...
+
+func waitForInstanceRunning(ctx context.Context, d *schema.ResourceData, meta interface{}, instanceID string, pending []string) error {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      pending,
+		Target:       []string{"RUNNING"},
+		Refresh:      ddmInstanceStatusRefreshFunc(instanceID, region, cfg),
+		Timeout:      d.Timeout(schema.TimeoutDelete),
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go]: the Timeout should not be fixed to TimeoutDelete

File: docs/resources/ddm_schema.md [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+
+* `create` - Default is 5 minute.
+* `delete` - Default is 30 minute.
+
+## Import
+
+The ddm schema can be imported using the `id`, e.g.
+
+```
+$ terraform import huaweicloud_ddm_schema.test 80e373f9-872e-4046-aae9-ccd9ddc55511/db_name
```
Comment (Jason-Zhang9309) [inline docs/resources/ddm_schema.md]:
```suggestion
$ terraform import huaweicloud_ddm_schema.test 80e373f9-872e-4046-aae9-ccd9ddc55511/schema_name
```

File: docs/resources/ddm_schema.md [hunk 6]
```diff
@@ -0,0 +1,183 @@
+---
+subcategory: "Distributed Database Middleware (DDM)"
+---
+
+# huaweicloud_ddm_schema
+
+Manages DDM schema resource within HuaweiCloud.
```
Comment (Jason-Zhang9309) [inline docs/resources/ddm_schema.md]:
```suggestion
Manages a DDM schema resource within HuaweiCloud.
```

File: docs/resources/ddm_schema.md [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+  less than or equal to the product of the associated RDS instances multiplied by 64.
+
+  Changing this parameter will create a new resource.
+
+* `data_nodes` - (Required, List, ForceNew) Specifies the RDS instances associated with the schema.
+
+  Changing this parameter will create a new resource.
+  The [DataNode](#DdmSchema_DataNode) structure is documented below.
+
+* `shard_unit` - (Optional, Int, ForceNew) Specifies the number of shards per RDS instance.
```
Comment (Jason-Zhang9309) [inline docs/resources/ddm_schema.md]: as discussed, this parameter is useless, please remove it

File: huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go (hunk 1 above)
Comment (houpeng80) [reply huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go]: done

File: huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go]: done

File: huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go (hunk 3 above)
Comment (houpeng80) [reply huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go]: done

File: huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go (hunk 4 above)
Comment (houpeng80) [reply huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go]: done

File: docs/resources/ddm_schema.md (hunk 5 above)
Comment (houpeng80) [reply docs/resources/ddm_schema.md]: done

File: docs/resources/ddm_schema.md (hunk 6 above)
Comment (houpeng80) [reply docs/resources/ddm_schema.md]: done

File: docs/resources/ddm_schema.md (hunk 7 above)
Comment (houpeng80) [reply docs/resources/ddm_schema.md]: done

File: docs/resources/ddm_schema.md [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+
+* `name` - (Required, String, ForceNew) Specifies the name of the DDM schema.
+  An instance name starts with a letter, consists of 2 to 48 characters, and can contain only lowercase letters,
+  digits, and underscores (_). Cannot contain keywords information_schema, mysql, performance_schema, or sys.
+
+  Changing this parameter will create a new resource.
+
+* `shard_mode` - (Required, String, ForceNew) Specifies the sharding mode of the schema. Values option: **cluster**, **single**.
+  **cluster**: indicates that the schema is in sharded mode.
+  **single**: indicates that the schema is in unsharded mode.
```
Comment (ShiChangkuo) [inline docs/resources/ddm_schema.md]:
```suggestion
  + **cluster**: indicates that the schema is in sharded mode.
  + **single**: indicates that the schema is in unsharded mode.
```

File: docs/resources/ddm_schema.md [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+## Timeouts
+
+This resource provides the following timeouts configuration options:
+
+* `create` - Default is 5 minutes.
+* `delete` - Default is 30 minutes.
+
+## Import
+
+The ddm schema can be imported using the `<instance_id>/<schema_name>`, e.g.
```
Comment (ShiChangkuo) [inline docs/resources/ddm_schema.md]:
```suggestion
The DDM schema can be imported using the `instance_id` and `schema_name`, separated by a slash, e.g.
```

File: huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go [hunk 10]
```diff
@@ -0,0 +1,477 @@
...
+	}
+	createSchemaOpt.JSONBody = utils.RemoveNil(buildCreateSchemaBodyParams(d))
+
+	var createSchemaResp *http.Response
+	for {
+		createSchemaResp, err = createSchemaClient.Request("POST", createSchemaPath, &createSchemaOpt)
+		if err == nil {
+			break
+		}
+		if errCode, ok := err.(golangsdk.ErrUnexpectedResponseCode); ok && errCode.Actual == 409 {
```
Comment (ShiChangkuo) [inline huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go:217]: it's better to add more comments

File: huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go [hunk 11]
```diff
@@ -0,0 +1,477 @@
...
+func waitForInstanceRunning(ctx context.Context, d *schema.ResourceData, cfg *config.Config, instanceID string,
+	pending []string, timeout string) error {
+	region := cfg.GetRegion(d)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      pending,
+		Target:       []string{"RUNNING"},
+		Refresh:      ddmInstanceStatusRefreshFunc(instanceID, region, cfg),
+		Timeout:      d.Timeout(timeout),
+		PollInterval: 5 * time.Second,
```
Comment (ShiChangkuo) [inline huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go:473]: Delay is missing?

File: docs/resources/ddm_schema.md (hunk 8 above)
Comment (houpeng80) [reply docs/resources/ddm_schema.md]: done

File: docs/resources/ddm_schema.md (hunk 9 above)
Comment (houpeng80) [reply docs/resources/ddm_schema.md]: done

File: huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go (hunk 10 above)
Comment (houpeng80) [reply huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go:217]: done

File: huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go (hunk 11 above)
Comment (houpeng80) [reply huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go:473]: done

File: huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go [hunk 12]
```diff
@@ -0,0 +1,478 @@
...
+			},
+			"admin_user": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the username for logging in to the associated RDS instance.`,
+			},
+			"admin_password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the password for logging in to the associated RDS instance.`,
```
Comment (ShiChangkuo) [inline huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go:137]: Missing Sensitive

Comment (houpeng80) [reply huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go:137]: done

**Pull request merged**